ensime_shared/launcher.py

```diff
@@ -163,6 +163,13 @@ def _start_process(self, classpath):
         null = open(os.devnull, "r")
         java = os.path.join(self.config['java-home'], 'bin', 'java')
 
+        if(os.name == "nt"):
+            java += ".exe" #the java executable has the .exe extension in windows
```

Comment:
I didn't like the `.exe` hack at first either, but now after some research and trying out the `java-home` directory existence test I think the `.exe` extension actually is the way to go.  Here is why:

1.  The `.exe`  addition is actually more robust on a windows system then leaving it off.  To make the auto `.exe` extention work, the users `PATHEXT` environment variable has to be properly setup with a `.exe` listing.
2. The `.exe` is required if we still want to test if the file is set as executable for the current user
3. The `java-home` directly only checks if that directory is availble, while it is unlikely the files below would be there it is not necessarily true that since the java_home path is often symlink'ed.